tests/Unit/Domain/Speaker/SpeakerProfileTest.php

```diff
@@ -0,0 +1,627 @@
+<?php
+
+namespace OpenCFP\Test\Unit\Domain\Speaker;
+
+use OpenCFP\Domain\Model;
+use OpenCFP\Domain\Speaker\NotAllowedException;
+use OpenCFP\Domain\Speaker\SpeakerProfile;
+use OpenCFP\Test\Helper\Faker\GeneratorTrait;
+use PHPUnit\Framework;
+
+/**
+ * @covers \OpenCFP\Domain\Speaker\SpeakerProfile
+ */
+final class SpeakerProfileTest extends Framework\TestCase
+{
+    use GeneratorTrait;
+
+    public function testIsAllowedToSeeReturnsFalseIfPropertyIsHidden()
+    {
+        $property = $this->getFaker()->word;
+
+        $hiddenProperties = [
+            $property,
+        ];
+
+        $profile = new SpeakerProfile(
+            $this->createUserMock(),
+            $hiddenProperties
+        );
+
+        $this->assertFalse($profile->isAllowedToSee($property));
+    }
+
+    public function testIsAllowedToSeeReturnsTrueIfPropertyIsNotHidden()
+    {
+        $property = $this->getFaker()->word;
+
+        $profile = new SpeakerProfile($this->createUserMock());
+
+        $this->assertTrue($profile->isAllowedToSee($property));
+    }
+
+    /**
+     * @dataProvider providerHasMadeProfile
+     *
+     * @param mixed $hasMadeProfile
+     */
+    public function testNeedsProfileReturnsFalseIfUserHasMadeProfile($hasMadeProfile)
+    {
+        $speaker = $this->createUserMock([
+            'has_made_profile' => $hasMadeProfile,
+        ]);
+
+        $profile = new SpeakerProfile($speaker);
+
+        $this->assertFalse($profile->needsProfile());
+    }
+
+    public function providerHasMadeProfile(): \Generator
```

Comment:
I'm using a data provider here (and in the test below) because we are using a non-strict comparison in [`SpeakerProfile::needsProfile()`](https://github.com/opencfp/opencfp/blob/40ce73ce014afec8faf92443401688ee8975d6b9/classes/Domain/Speaker/SpeakerProfile.php#L47-L50).

Ideally we would be using a strict comparison, and then the data provider would become obsolete.